pkg/utils/manager/manager.go

```diff
@@ -0,0 +1,60 @@
+package manager
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
```

Comment:
Can we add manager_test.go and unit tests for this newly added code ?

Reply:
done

Reply:
done